app/src/main/cpp/HandMeshRenderer.h

```diff
@@ -0,0 +1,61 @@
+/* -*- Mode: C++; tab-width: 20; indent-tabs-mode: nil; c-basic-offset: 2 -*-
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#pragma once
+
+#include "vrb/CreationContext.h"
+
+namespace crow {
+
+enum class HandMeshStrategy { None, Skinned, Spheres };
+
+struct Controller;
+
+class HandMeshRenderer;
+typedef std::shared_ptr<HandMeshRenderer> HandMeshRendererPtr;
+
+class HandMeshRenderer {
+protected:
+    vrb::CreationContextWeak context;
+    bool enabled = false;
+    virtual void Shutdown() { };
+public:
+    virtual void Initialize(vrb::CreationContextPtr& aContext) { context = aContext; };
+    virtual void Update(Controller& aController, const vrb::GroupPtr& aRoot, const bool aEnabled) = 0;
+    virtual void Draw(Controller& aController, const vrb::Camera& aCamera) { };
+};
+
+class HandMeshRendererSpheres: public HandMeshRenderer {
+protected:
+    struct State;
```

Comment:
This looks weird (protected vs private field)

Reply:
This is a common pattern in vrb (and some Wolvic) classes, but true that it can be all private.